<a href="https://colab.research.google.com/github/Mmbsaksd/Seoul-Bike-Sharing-Demand-Prediction/blob/main/Bike_Sharing_Demand_Prediction_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <b><u> Project Title : Seoul Bike Sharing Demand Prediction </u></b>

## <b> Problem Description </b>

### Currently Rental bikes are introduced in many urban cities for the enhancement of mobility comfort. It is important to make the rental bike available and accessible to the public at the right time as it lessens the waiting time. Eventually, providing the city with a stable supply of rental bikes becomes a major concern. The crucial part is the prediction of bike count required at each hour for the stable supply of rental bikes.


## <b> Data Description </b>

### <b> The dataset contains weather information (Temperature, Humidity, Windspeed, Visibility, Dewpoint, Solar radiation, Snowfall, Rainfall), the number of bikes rented per hour and date information.</b>


### <b>Attribute Information: </b>

* ### Date : year-month-day
* ### Rented Bike count - Count of bikes rented at each hour
* ### Hour - Hour of he day
* ### Temperature-Temperature in Celsius
* ### Humidity - %
* ### Windspeed - m/s
* ### Visibility - 10m
* ### Dew point temperature - Celsius
* ### Solar radiation - MJ/m2
* ### Rainfall - mm
* ### Snowfall - cm
* ### Seasons - Winter, Spring, Summer, Autumn
* ### Holiday - Holiday/No holiday
* ### Functional Day - NoFunc(Non Functional Hours), Fun(Functional hours)

#**Data Preprocessing**

In [23]:
#importing the essential libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#importing datetime moduel
import datetime

#importing warning
import warnings
warnings.filterwarnings("ignore")

In [4]:
#mounting the drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [13]:
#Loading the datasets
df = pd.read_csv('/content/SeoulBikeData.csv', encoding= 'unicode_escape')
df.head(24)

Date  Rented Bike Count  Hour  Temperature(°C)  Humidity(%)  \
0   01/12/2017                254     0             -5.2           37   
1   01/12/2017                204     1             -5.5           38   
2   01/12/2017                173     2             -6.0           39   
3   01/12/2017                107     3             -6.2           40   
4   01/12/2017                 78     4             -6.0           36   
5   01/12/2017                100     5             -6.4           37   
6   01/12/2017                181     6             -6.6           35   
7   01/12/2017                460     7             -7.4           38   
8   01/12/2017                930     8             -7.6           37   
9   01/12/2017                490     9             -6.5           27   
10  01/12/2017                339    10             -3.5           24   
11  01/12/2017                360    11             -0.5           21   
12  01/12/2017                449    12              1.7           23   
13  01/12/2017                451    13              2.4           25   
14  01/12/2017                447    14              3.0           26   
15  01/12/2017                463    15              2.1           36   
16  01/12/2017                484    16              1.2           54   
17  01/12/2017                555    17              0.8           58   
18  01/12/2017                862    18              0.6           66   
19  01/12/2017                600    19              0.0           77   
20  01/12/2017                426    20             -0.3           79   
21  01/12/2017                405    21             -0.8           81   
22  01/12/2017                398    22             -0.9           83   
23  01/12/2017                323    23             -1.3           84   

    Wind speed (m/s)  Visibility (10m)  Dew point temperature(°C)  \
0                2.2              2000                      -17.6   
1                0.8              2000                      -17.6   
2                1.0              2000                      -17.7   
3                0.9              2000                      -17.6   
4                2.3              2000                      -18.6   
5                1.5              2000                      -18.7   
6                1.3              2000                      -19.5   
7                0.9              2000                      -19.3   
8                1.1              2000                      -19.8   
9                0.5              1928                      -22.4   
10               1.2              1996                      -21.2   
11               1.3              1936                      -20.2   
12               1.4              2000                      -17.2   
13               1.6              2000                      -15.6   
14               2.0              2000                      -14.6   
15               3.2              2000                      -11.4   
16               4.2               793                       -7.0   
17               1.6              2000                       -6.5   
18               1.4              2000                       -5.0   
19               1.7              2000                       -3.5   
20               1.5              1913                       -3.5   
21               0.8              1687                       -3.6   
22               1.5              1380                       -3.4   
23               1.0              1265                       -3.6   

    Solar Radiation (MJ/m2)  Rainfall(mm)  Snowfall (cm) Seasons     Holiday  \
0                      0.00           0.0            0.0  Winter  No Holiday   
1                      0.00           0.0            0.0  Winter  No Holiday   
2                      0.00           0.0            0.0  Winter  No Holiday   
3                      0.00           0.0            0.0  Winter  No Holiday   
4                      0.00           0.0            0.0  

In [7]:
#Checking the type
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8760 entries, 0 to 8759
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Date                       8760 non-null   object 
 1   Rented Bike Count          8760 non-null   int64  
 2   Hour                       8760 non-null   int64  
 3   Temperature(°C)            8760 non-null   float64
 4   Humidity(%)                8760 non-null   int64  
 5   Wind speed (m/s)           8760 non-null   float64
 6   Visibility (10m)           8760 non-null   int64  
 7   Dew point temperature(°C)  8760 non-null   float64
 8   Solar Radiation (MJ/m2)    8760 non-null   float64
 9   Rainfall(mm)               8760 non-null   float64
 10  Snowfall (cm)              8760 non-null   float64
 11  Seasons                    8760 non-null   object 
 12  Holiday                    8760 non-null   object 
 13  Functioning Day            8760 non-null   objec

In [11]:
#Checking the null value
df.isnull().sum().sort_values(ascending=False)

Date                         0
Rented Bike Count            0
Hour                         0
Temperature(°C)              0
Humidity(%)                  0
Wind speed (m/s)             0
Visibility (10m)             0
Dew point temperature(°C)    0
Solar Radiation (MJ/m2)      0
Rainfall(mm)                 0
Snowfall (cm)                0
Seasons                      0
Holiday                      0
Functioning Day              0
dtype: int64

In [12]:
df.shape

(8760, 14)

**Observation on Preprocessing**

*   There are no null value in datasets
*   Some object value are there
*   Shape of the datasets is (8760, 14)
*   Here we can find time stamp



#**Feature Engineering**

In [19]:
#Extracting month from given datasets

month = []
for i in df['Date']:
  df_date = datetime.strptime(i,"%d-%m-%Y")
  month.append(df_date.month)

month
datee = datetime.datetime.strptime(a, "%Y-%m-%d")


ValueError: ignored

In [ ]:
# Pre processing train data
# Adding time and month to the dataframe using datatime column
time=[]
month=[]
for i in df['datetime']:
    dt_object2 = datetime.strptime(i, “%Y-%m-%d %H:%M:%S”)
    time.append(dt_object2.hour)
    month.append(dt_object2.month)
df['time'] = pd.DataFrame(time)
df['time'] = df['time'].astype(float)
df['month'] = pd.DataFrame(month)
df['month'] = df['month'].astype(float)

In [14]:
df.columns

Index(['Date', 'Rented Bike Count', 'Hour', 'Temperature(°C)', 'Humidity(%)',
       'Wind speed (m/s)', 'Visibility (10m)', 'Dew point temperature(°C)',
       'Solar Radiation (MJ/m2)', 'Rainfall(mm)', 'Snowfall (cm)', 'Seasons',
       'Holiday', 'Functioning Day'],
      dtype='object')

In [25]:
datee = datetime.datetime.strptime(df['Date'], "%Y-%m-%d")
datee.month


TypeError: ignored